### DISCUSSION TITLE
How to install entware at non /opt path, or compile source code modify /opt path to other path like /jun/opat

### DISCUSSION THREAD

**Discussion by mojun** [Entware/Entware#1074] [Q&A]
My router is readonly file system，but i want use entware .but no /opt path , 
How to modify source code, to make /opt to other path 
Please help me .

**Comment by zyxmon** [comment]
Here is instruction (Russian) to install Entware in any folder - https://web.archive.org/web/20230127183933/https://forums.zyxmon.org/viewtopic.php?f=5&t=5797
It is valid for armv7sf-k3.2

**Comment by you-oops-dev** [comment]
@zyxmon спасибо чувак телики ядренные делают где большинство папок в root директории на RO, а папки opt нет вообще.
![изображение](https://github.com/user-attachments/assets/6b76df2d-76c0-4744-a17d-de32411d88af)

**Comment by MortalM** [comment]
i have the same problem ,my device is xiaomiAX3000T and redmiAX6000,  they don't have /opt ，and i can not touch it, becasue / is read only system,  can we chose the install  /opt  to  other place , please!

**Reply by MortalM** [reply]
![image](https://github.com/user-attachments/assets/57f9e11d-2155-439f-a3f6-6e0096207e78)
![image](https://github.com/user-attachments/assets/835637eb-9dfd-4b9e-98ba-93f4ea113938)
two way to install ,failed

**Reply by you-oops-dev** [reply]
Need one directory of some sort **in the roo**t that is in **RW and not in tmpfs**. If there is no such directory, it is not possible to install Entware. That's my conclusion.

P.S Translator used.

**Reply by MortalM** [reply]
Can the developer let us customize the installation location, e.g. from /opt to /etc

**Comment by jbsilva** [comment]
In my case I do have a `/ opt` directory, but it's already in use and I prefer to keep Entware separated (/opt/entware would be fine).

**Reply by zyxmon** [reply]
> In my case I do have a `/ opt` directory, but it's already in use and I prefer to keep Entware separated (/opt/entware would be fine).

Just install entware in any dir and make this dir /opt in chroot environment. The only disadvantage is - you may change original /opt to something else in Entware environment

**Comment by ryzhovau** [comment]
> Can the developer let us customize the installation location, e.g. from /opt to /etc

It's out of project goal, sorry.
